pkg/applicationserver/io/packages/alcsync/v1/factory.go

```diff
@@ -0,0 +1,104 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alcsyncv1
+
+import (
+	"encoding/binary"
+	"time"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/gpstime"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	lorautil "go.thethings.network/lorawan-stack/v3/pkg/util/lora"
+)
+
+// CommandFactory is the factory for commands.
+type CommandFactory struct{}
+
+// NewCommandFactory returns a new CommandFactory.
+func NewCommandFactory() CommandFactory {
+	return CommandFactory{}
+}
+
+// newTimeSyncCommand builds a new TimeSyncCommand.
+func (CommandFactory) newTimeSyncCommand(
+	data []byte,
+	threshold uint32,
+	receivedAt time.Time,
+	fPort uint32,
+) (*TimeSyncCommand, []byte, error) {
+	// DeviceTime - bytes 0,1,2,3
```

Comment:
Nit: dot at the end. Also you can use range notation (`[0, 3]`) to save a bit of space.